Mage/src/main/java/mage/abilities/effects/common/combat/RedirectAttackerTargetEffect.java

```diff
@@ -0,0 +1,104 @@
+package mage.abilities.effects.common.combat;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.effects.OneShotEffect;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.filter.FilterPermanent;
+import mage.filter.StaticFilters;
+import mage.filter.predicate.Predicates;
+import mage.game.Game;
+import mage.game.combat.CombatGroup;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetDefender;
+
+public class RedirectAttackerTargetEffect extends OneShotEffect {
```

Comment:
I don't like naming it Redirect because that has a separate rules meaning. I suggest "ReselectAttackEffect", open to other suggestions as well

Reply:
Reselect sounds good to me, as it's the text the cards use. "You may reselect"